api/shareable.py

```diff
@@ -0,0 +1,55 @@
+from bson.objectid import ObjectId
+
+from pydantic import BaseModel
+from pymongo import MongoClient
+
+from loguru import logger
+
+
+class PlanData(BaseModel):
+    json_string: str
+    validation_levels: list[int]
+
+
+class MongoDBConnection:
+    def __init__(self):
+        self.connection = None
+
+    async def connect(self):
+        client = MongoClient("localhost", 27017)
+        db = client["plans"]
+
+        if "links" not in db.list_collection_names():
+            db.create_collection("links")
+            logger.info("Created 'links' collection")
+
+        self.collection = db["links"]
+        logger.success("MongoDB connection established successfully")
+
+    async def get_record(self, id):
+        record = self.collection.find_one({"_id": ObjectId(id)})
```

Comment:
What happens if no item is found?  Maybe a user mistypes the link or something?

Reply:
Now `alert` will be raised if an invalid ID(containing non-hexadecimal elements) is supplied or plan is not found for a particular ID.